### ISSUE TITLE
NIRCam WFSS schema is not backwards compatible

### ISSUE THREAD

**Issue by pllim** [spacetelescope/stdatamodels#159]
This works with stdatamodels 1.4.0 but fails in dev. This is from Jdaviz regression test suite and the data was provided by INS scientist.

```python
from astropy.io import fits
from astropy.utils.data import download_file

from stdatamodels import asdf_in_fits

filename = download_file('https://stsci.box.com/shared/static/d5k9z5j05dgfv6ljgie483w21kmpevni.fits', cache=True)
file_obj = fits.open(filename)
af = asdf_in_fits.open(file_obj)  # ERROR
```

A really long traceback that ends with this:
```
stdatamodels/jwst/transforms/converters/jwst_models.py:23, in NIRCAMGrismDispersionConverter.from_yaml_tree_transform(self, node, tag, ctx)
     20 def from_yaml_tree_transform(self, node, tag, ctx):
     21     from stdatamodels.jwst.transforms import models
---> 23     _fname = getattr(models, node["model_type"])
     24     return _fname(list(node['orders']),
     25                   list(node['lmodels']),
     26                   list(node['xmodels']),
   (...)
     30                   list(node['inv_ymodels']),
     31                   )

KeyError: 'model_type'
```

Suspected cause: https://github.com/spacetelescope/stdatamodels/pull/124

**Comment by tapastro** [comment]
The model for this dataset changed - the schema is not backwards-compatible with older processed data. Not sure how to proceed, but you are correct that #124 broke this for older datasets.

**Comment by pllim** [comment]
> Not sure how to proceed

Me neither. Is INS going to ask everyone to re-download such dataset?

**Comment by tapastro** [comment]
Perhaps some effort could be made to make it backwards compatible by introducing a new schema for the new converter?

**Comment by pllim** [comment]
🤷

**Comment by nden** [comment]
We are investigating adding support for backwards compatibility.

**Comment by braingram** [comment]
Fixed in #161
